modules/ztp-image-based-upgrade-with-backup.adoc

```diff
@@ -0,0 +1,229 @@
+. Create the configuration for the OADP Operator to access your Object storage instance.
+//TODO Check existing OADP docs for steps, link to them if applicable
+
+. Specify what you want to back up:
+//TODO decide how to present this info. Might be best to create a reference section with all the example CRs.
+
+`backup_acm_klusterlet.yaml`:: This allows you to reattach the cluster to the hub cluster once the upgrade and restore are completed.
+`backup_localvolume.yaml`:: This allows to restore the persistent volume that contains the `index.html` of your application.
+`backup_app.yaml`:: The information that must be backed up is different for each application. This CR specifies what needs to be backed up to recover the applications.
+
+. Define the restore order for the OADP Operator in the `Restore` CR by using the `lca.openshift.io/apply-wave` field:
++
+[source,yaml]
+----
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: acm-klusterlet
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "1"
+spec:
+  backupName:
+    acm-klusterlet
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: apache-app
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "3"
+spec:
+  backupName:
+    apache-app
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: localvolume
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "2"
+spec:
+  backupName:
+    localvolume
+----
+
+. Create a `ConfigMap` CR for the backup and restore information that the {lcao} uses during the upgrade.
+
+.. Create a `kustomization.yaml` that appends the information to a `ConfigMap` CR:
++
+[source,yaml]
+----
+configMapGenerator:
+- name: oadp-cm
+  namespace: openshift-adp
+  files:
+  - backup_acm_klusterlet.yaml
+  - backup_localvolume.yaml
+  - backup_app.yaml
+  - restore_acm_klusterlet.yaml
+  - restore_localvolumes.yaml
+  - restore_app.yaml
+----
+
+.. Create the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ kustomize build ./ -o OadpCm.yaml
+----
++
+.Example output
++
+[source,terminal]
+----
+kind: ConfigMap
+metadata:
+  name: oadp-cm-7gtgdhmdc5
+  namespace: openshift-adp
+[...]
+----
+
+.. Apply the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ oc apply -f OadpCm.yaml
+----
+
+. Append the `imageBasedUpgrade` CR with the backup and restore information:
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: upgrade
+spec:
+  stage: Idle #TODO might need to change or move these steps
+  seedImageRef:
+    version: 4.14.6
+    image: <seed_container_image>
+  oadpContent:
+  - name: oadp-cm-7gtgdhmdc5
+    namespace: openshift-adp
+----
+
+The OADP Operator creates a backup of the data specified in the OADP `Backup` and `Restore` CRs.
```

Comment:
Yes, but it will be created once we move to the Upgrade stage, which is the next step.